### ISSUE TITLE
Accept and decline buttons are grayed out after being kicked from a room and reinvited

### ISSUE THREAD

**Issue by aaronraimist** [neilalexander/seaglass#64]
## Specifications

  - Seaglass Version: 0.0.224-c536529 (actually 0.0.354?)
  - macOS Version: 10.13.6

## Expected Behavior

Accept and decline buttons on room invite should always work

## Actual Behavior

Accept and decline buttons on room invite are grayed out

## Steps to Reproduce the Problem

1. Create a new room from account 1
2. Invite account 2
3. Accept the room invite on account 2
4. Kick account 2 from the room
5. Invite account 2 to the room again

Workaround: quit Seaglass and launch again and the buttons will be enabled again

**Comment by neilalexander** [comment]
This should have been fixed in 1324d4fdffbdf206b3a083615b05ffbf57574a15 - can you please retest with the latest build?

**Comment by aaronraimist** [comment]
Will test this after #70 is fixed.

**Comment by aaronraimist** [comment]
Kicks don't work properly in 1324d4f, you continue see the room after the kick. It just says "Aaron Raimist has left the room". 

Then if you attempt to send a message you get this crash:
![screen shot 2018-09-12 at 9 03 46 pm](https://user-images.githubusercontent.com/5855073/45462872-5d069c80-b6cf-11e8-8363-6aa3042478d4.png)

**Comment by neilalexander** [comment]
Hopefully that's fixed in d2701e78cbdf31100d902a524d01b61b422b6fc9 now.

If the invite buttons look good then we should be OK to close the issue.

**Comment by neilalexander** [comment]
Have made some further changes to this behaviour in 05c23828a6264a08e35a123934caabd299f27b48 - the button enabling/disabling for all types of events is now handled by a single function which seems to be a bit more robust.

The label seems to be a bit more difficult - not sure why that's not rendering the right text. Can you retest the rest please?

**Comment by aaronraimist** [comment]
Looks good but invites don't show up until I relaunch Seaglass

**Comment by aaronraimist** [comment]
Actually I guess the invite does show up, just not at the top anymore and the input field and buttons didn't come back.

**Comment by neilalexander** [comment]
Yeah I'm having a hard time with the controls displaying reliably - I'm not sure if I'm just going cross-eyed from having been looking at the code for so long but from what I can tell this should work.

**Comment by aaronraimist** [comment]
Unfortunately not.

It might be worth me clearing the cache b/c I sometimes see the same room with an old name. There isn't an easy button hooked up to clear cache yet right?

Anyway `/me -> work` so I'll test any fixes you come up with later today.

**Comment by neilalexander** [comment]
Any luck with this on newer commits?

**Comment by aaronraimist** [comment]
So my raim.ist server is down at the moment so I found another unrelated bug which is preventing me from testing this. See #75

**Comment by neilalexander** [comment]
That should be fixed now.

**Comment by aaronraimist** [comment]
#75 is fixed but this issue is still present. After being kicked, accept/decline buttons still don't reappear until Seaglass is relaunched

**Comment by aaronraimist** [comment]
Sent a video in #seaglass since GitHub doesn't support .mov https://matrix.to/#/!qvCRFpZsGETXtHJSnJ:matrix.org/$1537222180115120qCQAQ:matrix.org
